src/lib/optimizer/strategy/join_to_predicate_rewrite_rule.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class PredicateNode;
```

Comment:
You don't need forward-declaration when you include the corresponding header files.